File: backend/model/saved_address.go
package model

import "gorm.io/gorm"

type SavedAddress struct {
	gorm.Model
	Name      string  `gorm:"not null"`
	UserID    uint    `gorm:"not null"`
	Address   string  `gorm:"not null"`
	Latitude  float64 `gorm:"not null"`
	Longitude float64 `gorm:"not null"`
	Notes     *string
}

type SavedAddressDTO struct {
	ID        uint    `json:"id"`
	Name      string  `json:"name"`
	Address   string  `json:"address"`
	Latitude  float64 `json:"latitude"`
	Longitude float64 `json:"longitude"`
	Notes     *string `json:"notes"`
}

type CreateSavedAddress struct {
	Name      string  `json:"name" binding:"required"`
	Address   string  `json:"address" binding:"required"`
	Latitude  float64 `json:"latitude" binding:"required"`
	Longitude float64 `json:"longitude" binding:"required"`
	Notes     *string `json:"notes"`
}

File: frontend/lib/model/saved_address.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'saved_address.freezed.dart';

part 'saved_address.g.dart';

@freezed
class SavedAddress with _$SavedAddress {
  const factory SavedAddress({
    required int id,
    required String name,
    required String address,
    required double latitude,
    required double longitude,
    required String? notes,
  }) = _SavedAddress;

  factory SavedAddress.fromJson(Map<String, dynamic> json) =>
      _$SavedAddressFromJson(json);
}
